core/vm/evm.go

```diff
@@ -529,16 +538,6 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		}
 	}
 
-	if err == nil && evm.chainRules.IsPrague {
-		if len(ret) > 0 {
-			touchCodeChunksRangeOnReadAndChargeGas(address.Bytes(), 0, uint64(len(ret)), uint64(len(ret)), evm.Accesses)
-		}
-		if !contract.UseGas(evm.Accesses.TouchAndChargeContractCreateCompleted(address.Bytes()[:])) {
-			evm.StateDB.RevertToSnapshot(snapshot)
-			err = ErrOutOfGas
-		}
-	}
```

Comment:
This code was moved before L534-L537, and note that L537 was deleted to avoid a double revert.